### PR TITLE
Change heavy-quark restart behavior

### PR DESCRIPTION
**Pull request by mathiaswagner**
This pull fixes:
* initialization of heavy-quark restart parameters
* disable mixed-precision solves for heavy-quarks as they are sometimes unstable (regular L2 which dominate are still using mixed precision)
* don't try to iterate L2 norm beyond machine precision when heavy-quark residual is requested

### CODE REVIEW

**Comment by mathiaswagner** [comment]
#854 to follow up for mixed precision.
Waiting for Carleton to confirm that current mixed precision workaround is fine.

File: lib/inv_cg_quda.cpp [hunk 1]
```diff
@@ -427,6 +427,7 @@ namespace quda {
     // set this to true if maxResIncrease has been exceeded but when we use heavy quark residual we still want to continue the CG
     // only used if we use the heavy_quark_res
     bool L2breakdown = false;
+    const double L2breakdown_eps = 100. * uhigh;
```
Comment (maddyscientist) [inline lib/inv_cg_quda.cpp]: Where does the factor of 100 come from?

Comment (mathiaswagner) [reply lib/inv_cg_quda.cpp]: Somewhat handwaving. We will never reach machine eps and while we might be able to get with in 10 * uhigh it seemed to tight for the purpose here. So just tried 100 ...

**Comment by mathiaswagner** [comment]
Yes. Only waiting on some feedback for mixed precision workaround.

**Comment by mathiaswagner** [comment]
Passed the MILC test :-)

**Pull request merged**